pkg/istio/rbac/common/util.go

```diff
@@ -99,15 +100,57 @@ func (a *ApiHandler) Delete(item *Item) error {
 	return err
 }
 
-// CheckIfMemberIsAllUsersFromDomain returns namespace for Athenz domain when role member is of form '<athenz-domain>.*'.
-// Example: domain.sub-domain.* -> domain-sub--domain
-func CheckIfMemberIsAllUsersFromDomain(member *zms.RoleMember, domainName zms.DomainName) (string, error) {
+// GetMemberName computes the name of the member based on if the
+// type of the member is *zms.GroupMember or *zms.RoleMember
+func GetMemberName(member interface{}) string {
+	if groupMember, ok := member.(*zms.GroupMember); ok {
+		return string(groupMember.MemberName)
+	}
+
+	if roleMember, ok := member.(*zms.RoleMember); ok {
+		return string(roleMember.MemberName)
+	}
+
+	return ""
+}
+
+// getMemberExpiry computes the expiry of the member based on if the
+// type of the member is *zms.GroupMember or *zms.RoleMember
+func getMemberExpiry(member interface{}) *rdl.Timestamp {
+	if groupMember, ok := member.(*zms.GroupMember); ok {
+		return groupMember.Expiration
+	}
+
+	if roleMember, ok := member.(*zms.RoleMember); ok {
+		return roleMember.Expiration
+	}
+
+	return &rdl.Timestamp{Time: time.Time{}}
+}
+
+// getMemberSystemDisabled computes is the system disabled flag
+// is enabled for a member based on if the type of the member
+// is *zms.GroupMember or *zms.RoleMember
+func getMemberSystemDisabled(member interface{}) *int32 {
+	if groupMember, ok := member.(*zms.GroupMember); ok {
+		return groupMember.SystemDisabled
+	}
 
+	if roleMember, ok := member.(*zms.RoleMember); ok {
+		return roleMember.SystemDisabled
+	}
+
+	return nil
```

Comment:
have the same question as above

Reply:
It should never be the case with our current setup. But this is in case we get something different and to handle that scenario.